fe/fe-core/src/main/java/org/apache/doris/nereids/rules/analysis/EliminateLogicalSelectHint.java

```diff
@@ -85,7 +90,15 @@ private void setVar(SelectHintSetVar selectHint, StatementContext context) {
         SessionVariable sessionVariable = context.getConnectContext().getSessionVariable();
         // set temporary session value, and then revert value in the 'finally block' of StmtExecutor#execute
         sessionVariable.setIsSingleSetVar(true);
-        for (Entry<String, Optional<String>> kv : selectHint.getParameters().entrySet()) {
+        Map<String, Optional<String>> parameters = selectHint.getParameters();
+        StatementBase parsedStatement = context.getParsedStatement();
+        if (parsedStatement instanceof LogicalPlanAdapter) {
+            LogicalPlan logicalPlan = ((LogicalPlanAdapter) parsedStatement).getLogicalPlan();
```

Comment:
should not impl here. please refer to `org.apache.doris.nereids.processor.pre.TurnOffPipelineForDml`